pkg/cmd/dedicated/deregister/deregister.go

```diff
@@ -0,0 +1,277 @@
+)
+
+func NewDeRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "deregister-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.deregisterCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.deregisterCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.deregisterCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runDeRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.deregisterCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.deregistercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.deregisterCluster.flag.clusterId.description"))
+
+	return cmd
+}
+
+func runDeRegisterClusterCmd(opts *options) (err error) {
+	// Set the base URL for the cluster management API
+	clusterList, err := getListOfClusters(opts)
+	if err != nil {
+		return err
+	}
+
+	if len(clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.deregisterCluster.run.noClusterFound")
+	}
+
+	// TO-DO if client has supplied a cluster id, validate it and set it as the selected cluster without listing getting all clusters
+	if opts.selectedClusterId == "" {
+		err = runClusterSelectionInteractivePrompt(opts, &clusterList)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range clusterList {
+			cluster := clusterList[i]
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
+	}
+
+	kafkas, err := getkafkasInCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	opts.f.Logger.Info(opts.f.Localizer.MustLocalize("dedicated.deregisterCluster.kafka.delete.warning"))
+
+	err = deleteKafkasPrompt(opts, &kafkas)
+	if err != nil {
+		return err
+	}
+
+	err = deregisterClusterFromKasFleetManager(opts)
+	if err != nil {
+		return err
+	}
+
+	addonIdsToDelete := []string{fleetshardAddonIdQE, fleetshardAddonId, strimziAddonId, strimziAddonIdQE}
+	err = ocmUtils.RemoveAddonsFromCluster(opts.f, opts.clusterManagementApiUrl, opts.accessToken, &opts.selectedCluster, addonIdsToDelete)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getListOfClusters(opts *options) ([]clustersmgmtv1.Cluster, error) {
+	clusters, err := ocmUtils.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, 1, 99)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []clustersmgmtv1.Cluster
+	for _, cluster := range clusters.Slice() {
+		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
+			list = append(list, *cluster)
+		}
+	}
+
+	return list, nil
+}
```

Comment:
This will enlist all the clusters even the ones that are not registered in KFM.

My suggestion will be to only enlist the clusters that are registered within KFM.

i.e a workflow like: 

1. make the call to KFM to retrieve the list of registered clusters
2. make the call to OCM and get the clusters names only for the clusters that have been registered within KFM
3. Display the cluster name 

Just like for the create interactive command.

Reply:
Okay will try now

Reply:
Added seems to be working and no need to hardcode sizes anymore 🥳